src/middlewared/middlewared/etc_files/smb_configure.py

```diff
@@ -297,4 +297,5 @@ def render(service, middleware):
     if conf['passdb_backend'] == "tdbsam":
         middleware.call_sync('smb.synchronize_passdb')
         validate_group_mappings(middleware, conf)
+        middleware.call_sync('smb.check_rid_conflict')
```

Comment:
Do we want to wait for this to complete?

Reply:
No. It may take a long time to complete if there are _huge_ numbers of groups and it won't interfere with other things we are doing. So intention is to background it.